pyogrio/geopandas.py

```diff
@@ -99,11 +103,17 @@ def read_dataframe(
         where=where,
         bbox=bbox,
         fids=fids,
+        return_fids=set_fid_index,
     )
 
     columns = meta["fields"].tolist()
     data = {columns[i]: field_data[i] for i in range(len(columns))}
-    df = pd.DataFrame(data, columns=columns)
+    if set_fid_index:
+        index = index = pd.Series(index, name="fid")
```

Comment:
Can we create `pandas.Index` here directly instead of going via `Series`? Do we know the `dtype`? Can we pass it or we just let pandas figure it out?
```suggestion
        index = pd.Index(index, name="fid")
```

Reply:
Incoming values should be `int64` type, so I think we can use `index = pd.Int64Index(index, name="fid")` to be explicit about that.

Reply:
`pd.Int64Index` will be deprecated in the future, so you can use just `pd.Index`. I suppose the data is already a numpy array of type int64? In that case, both are exactly equivalent and will preserve the dtype of the array.